hudi-common/src/avro/test/java/org/apache/parquet/schema/TestAvroSchemaRepair.java

```diff
@@ -0,0 +1,984 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.schema;
+
+import org.apache.hudi.avro.AvroSchemaUtils;
+
+import org.apache.avro.LogicalTypes;
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaBuilder;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotSame;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests {@link AvroSchemaRepair}.
+ */
+public class TestAvroSchemaRepair {
+
+  @Test
+  public void testNoRepairNeededIdenticalSchemas() {
+    Schema requestedSchema = Schema.create(Schema.Type.LONG);
+    Schema tableSchema = Schema.create(Schema.Type.LONG);
+
+    Schema result = AvroSchemaRepair.repairLogicalTypes(requestedSchema, tableSchema);
+
+    assertSame(requestedSchema, result, "When schemas are identical, should return same instance");
+
+  }
+
+  @Test
+  public void testNoRepairNeededDifferentPrimitiveTypes() {
+    Schema requestedSchema = Schema.create(Schema.Type.STRING);
+    Schema tableSchema = Schema.create(Schema.Type.INT);
+
+    Schema result = AvroSchemaRepair.repairLogicalTypes(requestedSchema, tableSchema);
+
+    assertSame(requestedSchema, result, "When types differ, should return original schema");
+  }
+
+  @Test
+  public void testRepairLongWithoutLogicalTypeToLocalTimestampMillis() {
+    Schema requestedSchema = Schema.create(Schema.Type.LONG);
+    Schema tableSchema = LogicalTypes.localTimestampMillis().addToSchema(Schema.create(Schema.Type.LONG));
+
+    Schema result = AvroSchemaRepair.repairLogicalTypes(requestedSchema, tableSchema);
+
+    assertNotSame(requestedSchema, result, "Should create a new schema with logical type");
+    assertEquals(Schema.Type.LONG, result.getType());
+    assertEquals(LogicalTypes.localTimestampMillis(), result.getLogicalType());
+  }
+
+  @Test
+  public void testRepairLongWithoutLogicalTypeToLocalTimestampMicros() {
+    Schema requestedSchema = Schema.create(Schema.Type.LONG);
+    Schema tableSchema = LogicalTypes.localTimestampMicros().addToSchema(Schema.create(Schema.Type.LONG));
+
+    Schema result = AvroSchemaRepair.repairLogicalTypes(requestedSchema, tableSchema);
+
+    assertNotSame(requestedSchema, result, "Should create a new schema with logical type");
+    assertEquals(Schema.Type.LONG, result.getType());
+    assertEquals(LogicalTypes.localTimestampMicros(), result.getLogicalType());
+  }
```

Comment:
Should we repairing in this scenario? What if actual file logical type is millis but it is not present in the schema?